src/components/Comments.jsx

```diff
@@ -1,19 +1,39 @@
-import { useEffect, useState } from 'react';
-import { getArticleCommentsByID } from '../utils/api';
+import { useEffect, useState, useContext } from 'react';
+import { getArticleCommentsByID, deleteCommentByID } from '../utils/api';
 import PostComment from './PostComment';
 import { useParams } from 'react-router-dom';
+import { UserContext } from '../contexts/User';
+import CommentVotes from './CommentVotes';
 
 const Comments = () => {
+	const { user, isLoggedIn } = useContext(UserContext);
 	const [comments, setComments] = useState([]);
 	const { article_id } = useParams();
 	const [isLoading, setIsLoading] = useState(true);
+	const [disableButton, setDisableButton] = useState(false);
 
 	useEffect(() => {
 		getArticleCommentsByID(article_id).then(({ commentData }) => {
 			setComments(commentData);
 			setIsLoading(false);
 		});
 	});
+
+	const deleteComment = (comment_id) => {
```

Comment:
This works fine and you have chose to optimistically render the delete. If the delete is unsuccessful we are currently unable to re-render the comments back. I would suggest not using optimistic rendering here and only update the UI once you get to the then block.